Modules/Office365DSC/DSCResources/MSFT_EXOAntiPhishRule/MSFT_EXOAntiPhishRule.psm1

```diff
@@ -0,0 +1,487 @@
+        }
+
+    }
+
+    if ( ('Absent' -eq $Ensure ) -and ($AntiPhishRule) )
+    {
+        Write-Verbose "Removing AntiPhishRule $($Identity) "
+        try
+        {
+            Remove-AntiPhishRule -Identity $Identity -Confirm:$false
+        }
+        catch
+        {
+            Close-SessionsAndReturnError -ExceptionMessage $_.Exception
+        }
+
+    }
+    Write-Verbose "Closing Remote PowerShell Sessions"
+    $ClosedPSSessions = (Get-PSSession | Remove-PSSession)
+    Write-Verbose "Global ExchangeOnlineSession status: `n"
+    Write-Verbose "$( Get-PSSession -ErrorAction SilentlyContinue | Where-Object Name -eq 'ExchangeOnline' | Out-String)"
+
+}
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $AntiPhishPolicy,
+
+        [Parameter()]
+        [System.String]
+        $Comments,
+
+        [Parameter()]
+        [System.Boolean]
+        $Enabled = $true,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.String[]]
+        $ExceptIfRecipientDomainIs = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $ExceptIfSentTo = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $ExceptIfSentToMemberOf = @(),
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Identity,
+
+        [Parameter()]
+        [uint32]
+        $Priority,
+
+        [Parameter()]
+        [System.String[]]
+        $RecipientDomainIs = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $SentTo = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $SentToMemberOf = @()
+
+    )
+    Write-Verbose -Message "Testing AntiPhishRule for $($Identity)"
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+    $AntiPhishRuleTestParams = $PSBoundParameters
+    $AntiPhishRuleTestParams.Remove("GlobalAdminAccount") | out-null
+    $TestResult = Test-Office365DSCParameterState -CurrentValues $CurrentValues `
+        -DesiredValues $PSBoundParameters `
+        -ValuesToCheck $AntiPhishRuleTestParams.Keys
+    if ($TestResult)
+    {
+        Write-Verbose "Closing Remote PowerShell Sessions"
+        $ClosedPSSessions = (Get-PSSession | Remove-PSSession)
+        Write-Verbose "Global ExchangeOnlineSession status: `n"
+        Write-Verbose "$( Get-PSSession -ErrorAction SilentlyContinue | Where-Object Name -eq 'ExchangeOnline' | Out-String)"
+    }
+
+    return $TestResult
+}
+
+function Export-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.String])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $AntiPhishPolicy,
```

Comment:
Only mandatory parameters are expected in the function's signature

Reply:
I'm not sure I understand what you're saying here. Are you saying that the Export-TargetResource functions only need to have mandatory parameters in the param block?

Reply:
Removed non-Mandatory parameters from Export-TargetResource functions.